### PR TITLE
wokers: api-server: admin: decimal-adjust fillable price

### PR DESCRIPTION
**Pull request by akirillo**
This PR fixes the calculation of the fillable amount in the admin order metadata endpoint. We need to decimal-correct the price when computing the fillable amount, but we return the "original" price (which effectively assumes the base/quote tokens use the same number of decimals) to the client, as this is easier to interpret. This basically defers the responsibility of decimal-adjusting the fillable amount to the client if it wishes to compute the value by multiplying it with the price.

### CODE REVIEW

File: common/src/types.rs [hunk 1]
```diff
@@ -57,6 +58,26 @@ impl TimestampedPrice {
     pub fn as_fixed_point(&self) -> FixedPoint {
         FixedPoint::from_f64_round_down(self.price)
     }
+
+    /// Decimal correct a price for a given token pair
+    pub fn decimal_correct(
```
Comment (joeykraut) [inline common/src/types.rs:63]: nit: `get_decimal_corrected_price` would be a better name here. The current name seems to imply that we mutate the receiver

File: workers/api-server/src/http/admin.rs [hunk 2]
```diff
@@ -551,19 +551,29 @@ async fn get_fillable_amount_and_price(
...
+            base_token: base_token.clone(),
+            quote_token: quote_token.clone(),
+            channel: price_tx,
+        })
         .map_err(internal_error)?;
 
-    let price = match price_rx.await.map_err(internal_error)? {
-        PriceReporterState::Nominal(report) => report.price,
+    let price: TimestampedPrice = match price_rx.await.map_err(internal_error)? {
+        PriceReporterState::Nominal(report) => (&report).into(),
```
Comment (joeykraut) [inline workers/api-server/src/http/admin.rs:562]: why this change? An opaque `into` is less readable than field access

**Review by joeykraut — APPROVED** [review]

File: workers/api-server/src/http/admin.rs (hunk 2 above)
Comment (akirillo) [reply workers/api-server/src/http/admin.rs:562]: so that we can call `TimestampedPrice::decimal_correct` on the result
